params/protocol_params.go

```diff
@@ -146,13 +146,43 @@ func GetMaximumExtraDataSize() uint64 {
 	return BFTMaximumExtraDataSize
 }
 
-type CodeFormat uint8
+// CodeInfo consists of 8 bits, and has information of the contract code.
+// Originally, codeInfo only contains codeFormat information(interpreter version), but now it is divided into two parts.
+// First four bit contains the deployment time (ex. 0x00(constantinople), 0x10(istanbul,...)), so it is called vmVersion.
+// Last four bit contains the interpreter version (ex. 0x00(EVM), 0x01(EWASM)), so it is called codeFormat.
+type CodeInfo uint8
+
+const (
+	// codeFormatBitMask filters only the codeFormat. It means the interpreter version used by the contract.
+	// Mask result 1. [x x x x 0 0 0 1]. The contract uses EVM interpreter.
+	codeFormatBitMask = 0b00001111
+
+	// vmVersionBitMask filters only the vmVersion. It means deployment time of the contract.
+	// Mask result 1. [0 0 0 0 x x x x]. The contract is deployed at constantinople
+	// Mask result 2. [0 0 0 1 x x x x]. The contract is deployed after istanbulHF
+	vmVersionBitMask = 0b11110000
+)
+
+func (t CodeInfo) GetCodeFormat() CodeFormat {
+	return CodeFormat(t & codeFormatBitMask)
+}
+
+func (t CodeInfo) GetVmVersion() VmVersion {
+	return VmVersion(t & vmVersionBitMask)
```

Comment:
How about using shift operator?
I think 0x01 is better than 0x10 for version 1.

```suggestion
	return VmVersion(t & vmVersionBitMask >> 4)
```

Reply:
Thanks for the comment! I changed it as you suggested. https://github.com/klaytn/klaytn/pull/912/commits/f21a7293d0e66c538653f3713661fa1c0d7a6a49